File: ts/src/vector/constant/intConstVector.ts
import type BitVector from "../flat/bitVector";
import Vector from "../vector";

export class IntConstVector extends Vector<Int32Array | Uint32Array, number> {
    public constructor(
        name: string,
        value: number,
        sizeOrNullabilityBuffer: number | BitVector,
        isSigned: boolean = true,
    ) {
        super(name, isSigned ? Int32Array.of(value) : Uint32Array.of(value), sizeOrNullabilityBuffer);
    }

    protected getValueFromBuffer(_index: number): number {
        return this.dataBuffer[0];
    }
}

File: java/src/main/java/org/maplibre/mlt/vector/constant/IntConstVector.java
package org.maplibre.mlt.vector.constant;

import org.maplibre.mlt.vector.BitVector;
import java.nio.IntBuffer;

public class IntConstVector extends ConstVector<IntBuffer, Integer> {

  public IntConstVector(String name, Integer value, int size) {
    super(name, IntBuffer.wrap(new int[] {value}), size);
  }

  public IntConstVector(String name, BitVector nullabilityBuffer, Integer value) {
    super(name, nullabilityBuffer, IntBuffer.wrap(new int[] {value}));
  }

  @Override
  protected Integer getValueFromBuffer(int index) {
    return dataBuffer.get(0);
  }
}
